application/core/LSMessageSource.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/**
+ * LSMessageSource class file.
+ *
+ * @author Denis Chenu
+ * @link http://www.yiiframework.com/
+ * @copyright 2021 LimeSurvey
+ * License: GNU/GPL License v3 or later, see LICENSE.php
+ */
+
+class LSMessageSource extends CMessageSource
+{
+    public const CACHE_KEY_PREFIX = 'Yii.LSMessageSource.';
+    public const MO_FILE_EXT = '.mo';
+    public const PO_FILE_EXT = '.po';
+    /**
+     * @var string the ID of the database connection application component. Defaults to 'db'.
+     * @see CDbMessageSource::connectionID
+     */
+    public $connectionID = 'db';
+    /**
+     * @var string the name of the source message table. Defaults to 'SourceMessage'.
+     * The DB prefix is added automatically
+     * @see CDbMessageSource::sourceMessageTable
+     */
+    public $sourceMessageTable = 'sourcemessage';
+    /**
+     * @var string the name of the translated message table. Defaults to 'Message'.
+     * The DB prefix is added automatically
+     * @see CDbMessageSource::translatedMessageTable
+     */
+    public $translatedMessageTable = 'message';
+    /**
+     * @var integer the time in seconds that the messages can remain valid in cache.
+     * @see CDbMessageSource::cachingDuration
+     * @see CGettextMessageSource::cachingDuration
+     */
+    public $cachingDuration = 0;
+    /**
+     * @var string the ID of the cache application component that is used to cache the messages.
+     * Defaults to 'cache' which refers to the primary cache application component.
+     * @see CDbMessageSource::cacheID
+     * @see CGettextMessageSource::cacheID
+     */
+    public $cacheID = 'cache';
+    /**
+     * @var string the base path for all translated messages. Defaults to null, meaning
+     * @see CGettextMessageSource::basePath
+     */
+    public $basePath;
```

Comment:
Same than db, table name etc … LimeSurvey offer is own system.

Did we allow to update basePath in config , then user can create their own po file.

Reply:
Leave it : didn't broke potential usage